### ISSUE TITLE
Demo code causes kernel crash in Jupyter in VS Code

### ISSUE THREAD

**Issue by alexgleith** [ec-jrc/nrt#10]
I'm testing this library, and the example from the readme crashes in VS Code.

This is the line that causes the crash: `EwmaMonitor = EWMA(trend=False, mask=mask)`

~The error is kind of hidden, because someone the kernel crashes, rather than Python throwing an exception. The error in the Jupyter logs is:~

~> NameError: name 'data' is not defined~

Edit: the above was unrelated. See below.

Let me know if you'd like to see more info about my Python environment.

**Comment by alexgleith** [comment]
Reproduced again with this code:

```python
from nrt.monitor.ewma import EWMA
from nrt import data

# load example mask
mask = (data.romania_forest_cover_percentage() > 20).astype('int')

# Instantiate
nrt_class = EWMA(
    mask=mask,
    trend=True,
    harmonic_order=3,
    sensitivity=2
)

# load example xarray
s2_cube = data.romania_20m()
history = s2_cube.B3.sel(time=slice(None, '2019-01-01'))
monitor = s2_cube.B3.sel(time=slice('2019-01-01', None))

# Fitting
nrt_class.fit(dataarray=history)

# Dump model
nrt_class.to_netcdf('model.nc')
```

It crashes on the `.fit()` line.

The crash is strange, as it makes the kernel crash.

<img width="1310" alt="image" src="https://github.com/ec-jrc/nrt/assets/3445853/0fea9327-5b91-4a30-b7d1-e2d949482baa">

**Comment by alexgleith** [comment]
Actually, the error reported at the top of the issue may be misleading. The logs don't actually show an error.

The crash log is like this: `19:42:02.704 [error] Disposing session as kernel process died ExitCode: undefined, Reason: `

**Comment by loicdtx** [comment]
Okay, thanks again for testing and reporting @alexgleith 

I'm looking into this right now. I could not reproduce using python 3.12.3 repl in a fresh docker container. Will setup a quick environment with 3.11 + jupyter to match exactly your environment

**Comment by alexgleith** [comment]
My environment is pretty insane... I should probably test in a simpler space :-) 

[pip freeze is here](https://gist.github.com/alexgleith/736a790093f029d1ffbf677d93b26359)

**Comment by loicdtx** [comment]
I was not able to reproduce using a similar environment

```Dockerfile
FROM python:3.11.9
RUN pip install --no-cache-dir jupyter
WORKDIR /workspace
EXPOSE 8888
CMD ["jupyter", "notebook", "--ip=0.0.0.0", "--port=8888", "--no-browser", "--allow-root"]
```

which results in the following list of python packages:

<details>

```bash
affine==2.4.0
anyio==4.4.0
argon2-cffi==23.1.0
argon2-cffi-bindings==21.2.0
arrow==1.3.0
asttokens==2.4.1
async-lru==2.0.4
attrs==23.2.0
Babel==2.15.0
beautifulsoup4==4.12.3
bleach==6.1.0
certifi==2024.6.2
cffi==1.16.0
cftime==1.6.4
charset-normalizer==3.3.2
click==8.1.7
click-plugins==1.1.1
cligj==0.7.2
comm==0.2.2
debugpy==1.8.1
decorator==5.1.1
defusedxml==0.7.1
executing==2.0.1
fastjsonschema==2.20.0
fqdn==1.5.1
h11==0.14.0
httpcore==1.0.5
httpx==0.27.0
idna==3.7
ipykernel==6.29.4
ipython==8.25.0
ipywidgets==8.1.3
isoduration==20.11.0
jedi==0.19.1
Jinja2==3.1.4
json5==0.9.25
jsonpointer==3.0.0
jsonschema==4.22.0
jsonschema-specifications==2023.12.1
jupyter==1.0.0
jupyter-console==6.6.3
jupyter-events==0.10.0
jupyter-lsp==2.2.5
jupyter_client==8.6.2
jupyter_core==5.7.2
jupyter_server==2.14.1
jupyter_server_terminals==0.5.3
jupyterlab==4.2.2
jupyterlab_pygments==0.3.0
jupyterlab_server==2.27.2
jupyterlab_widgets==3.0.11
llvmlite==0.43.0
MarkupSafe==2.1.5
matplotlib-inline==0.1.7
mistune==3.0.2
nbclient==0.10.0
nbconvert==7.16.4
nbformat==5.10.4
nest-asyncio==1.6.0
netCDF4==1.7.1
notebook==7.2.1
notebook_shim==0.2.4
nrt==0.2.0
numba==0.60.0
numpy==2.0.0
overrides==7.7.0
packaging==24.1
pandas==2.2.2
pandocfilters==1.5.1
parso==0.8.4
pexpect==4.9.0
platformdirs==4.2.2
prometheus_client==0.20.0
prompt_toolkit==3.0.47
psutil==6.0.0
ptyprocess==0.7.0
pure-eval==0.2.2
pycparser==2.22
Pygments==2.18.0
pyparsing==3.1.2
python-dateutil==2.9.0.post0
python-json-logger==2.0.7
pytz==2024.1
PyYAML==6.0.1
pyzmq==26.0.3
qtconsole==5.5.2
QtPy==2.4.1
rasterio==1.3.10
referencing==0.35.1
requests==2.32.3
rfc3339-validator==0.1.4
rfc3986-validator==0.1.1
rpds-py==0.18.1
scipy==1.13.1
Send2Trash==1.8.3
six==1.16.0
sniffio==1.3.1
snuggs==1.4.7
soupsieve==2.5
stack-data==0.6.3
terminado==0.18.1
tinycss2==1.3.0
tornado==6.4.1
traitlets==5.14.3
types-python-dateutil==2.9.0.20240316
typing_extensions==4.12.2
tzdata==2024.1
uri-template==1.3.0
urllib3==2.2.2
wcwidth==0.2.13
webcolors==24.6.0
webencodings==0.5.1
websocket-client==1.8.0
widgetsnbextension==4.0.11
xarray==2024.6.0
```
</details>

I usually suspect numba when this kind of kernel crashes happen. Your version (`0.57.1`) is a bit behind but not excessively old (May 2023). If you manage to upgrade numba without breaking everything else, it may give us a clue.

**Comment by alexgleith** [comment]
Ok, tried with new `numba`, now on version `0.60.0`. I tested using a jupyter notebook server, instead of VS Code too, but they both still had the kernel crash the same.

I just upgraded `numpy` though, as I had previously downgraded it for another project, and now the code runs!

So it was my weird environment, and seems to have been `numpy` in particular.

(Actually, it's more complicated than this... I think we leave it there though, at semi-mysterious, but "works"!)

I'll go update the review :-)
